sql/parser/builtins.go

```diff
@@ -1161,6 +1186,31 @@ func regexpEvalFlags(pattern, sqlFlags string) (string, error) {
 	return fmt.Sprintf("(?%s:%s)", flagString.String(), pattern), nil
 }
 
+func overlay(s, to string, pos, size int) (Datum, error) {
+	if pos < 1 {
+		return nil, errors.New("negative substring length not allowed")
+	}
+
+	var out bytes.Buffer
+	i := 0
+	for _, c := range s {
```

Comment:
Take a look at the implementation of "left" - you can `runes := []rune(s)` and then do roughly what I described. I'm not totally sure that's better, though.